packages/generator/src/complex-type/field-type-class.ts

```diff
@@ -96,5 +96,9 @@ export function createPropertyFieldInitializer(prop: VdmProperty): string {
     return `this._fieldBuilder.buildComplexTypeField('${prop.originalName}', ${prop.fieldType}, ${prop.nullable})`;
   }
 
+  if (prop.isEnum) {
```

Comment:
Do you need another branch for enum collection?

Reply:
ok, this was missing from the beginning and I fixed it.
It's like you have you have the following boolean, where every boolean might be true or false.
- isEnum
- isCollection
- isComplex